tap_snapchat_ads/streams.py

```diff
@@ -17,758 +17,1419 @@
-        'targeting_type': 'dlxp',
-        'key_properties': ['id'],
-        'replication_method': 'FULL_TABLE',
-        'path': 'targeting/{targeting_group}/{targeting_type}',
-        'data_key_array': 'targeting_dimensions',
-        'data_key_record': '{targeting_type}',
-        'api_method': 'GET',
-        'paging': True,
-        'params': {}
-    },
-    # INTERESTS - NLN (Nielsen)
-    # Reference: https://developers.snapchat.com/api/docs/#get-nielsen-interest-targeting-options
-    'targeting_interests_nln': {
-        'json_schema': 'schemas/shared/targeting.json',
-        'targeting_group': 'interests',
-        'targeting_type': 'nln',
-        'key_properties': ['id'],
-        'replication_method': 'FULL_TABLE',
-        'path': 'targeting/{targeting_group}/{targeting_type}',
-        'data_key_array': 'targeting_dimensions',
-        'data_key_record': '{targeting_type}',
-        'api_method': 'GET',
-        'paging': True,
-        'params': {}
-    },
-    # INTERESTS - PLC (Placed Visitation Segments)
-    # Reference: https://developers.snapchat.com/api/docs/#interests-placed-visitation-segments
-    'targeting_interests_plc': {
-        'json_schema': 'schemas/shared/targeting.json',
-        'targeting_group': 'interests',
-        'targeting_type': 'plc',
-        'key_properties': ['id'],
-        'replication_method': 'FULL_TABLE',
-        'path': 'targeting/{targeting_group}/{targeting_type}',
-        'data_key_array': 'targeting_dimensions',
-        'data_key_record': '{targeting_type}',
-        'api_method': 'GET',
-        'paging': True,
-        'params': {}
-    },
-
-    # LOCATION - Categories LOI
-    # Reference: https://developers.snapchat.com/api/docs/#location
-    'targeting_location_categories': {
-        'json_schema': 'schemas/shared/targeting.json',
-        'targeting_group': 'location',
-        'targeting_type': 'categories_loi',
-        'key_properties': ['id'],
-        'replication_method': 'FULL_TABLE',
-        'path': 'targeting/{targeting_group}/{targeting_type}',
-        'data_key_array': 'targeting_dimensions',
-        'data_key_record': '{targeting_type}',
-        'api_method': 'GET',
-        'paging': True,
-        'params': {}
+LOGGER = singer.get_logger()
+BASE_URL = 'https://adsapi.snapchat.com/v1'
+
+class SnapchatAds:
+    tap_stream_id = None
+    key_properties = []
+    replication_method = None
+    replication_keys = []
+    path = None
+    data_key_array = None
+    data_key_record = ""
+    paging = False
+    params = {}
+    parent_stream = None
+    parent = None
+    grandparent_stream = None
+    great_grandparent_stream = None
+    json_schema = None
+    bookmark_query_field_from = None
+    bookmark_query_field_to = None
+    date_window_size = 1
+    targeting_country_ind = False
+    targeting_group = None
+    targeting_type = None
+    children = []
+
+    # To write schema in output
+    def write_schema(self, catalog, stream_name):
```

Comment:
Add function comments for all the added functions.

Reply:
Added doc string for all newly added functions